.github/workflows/node-build.yml

```diff
@@ -0,0 +1,309 @@
+      - name: Save docker image to cache
+        uses: actions/cache@v4
+        with:
+          path: /tmp/${{ matrix.package }}.tar
+          key: ${{ github.sha }}
+  
+  docker-grype:
+    name: Docker Grype Scan
+    needs: docker-build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        package: [auth, backend, frontend]
+    steps:
+      - name: Fetch docker image from cache
+        uses: actions/cache/restore@v4
+        with:
+          path: /tmp/${{ matrix.package }}.tar
+          key: ${{ github.sha }}
+      - name: Scan docker image
+        uses: anchore/scan-action@v3
+        with:
+          image: /tmp/${{ matrix.package }}.tar
+          fail-build: true
+          only-fixed: true
+          severity-cutoff: high
+          output-format: table
+
+  docker-trivy:
+    name: Docker Trivy Scan
+    needs: [docker-build]
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        package: [auth, backend, frontend]
+    steps:
+      - name: Fetch docker image from cache
+        uses: actions/cache/restore@v4
+        with:
+          path: /tmp/${{ matrix.package }}.tar
+          key: ${{ github.sha }}
+      - name: Download Trivy
+        run: |
+          curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /tmp
+      - name: Scan docker image
+        run: |
+          /tmp/trivy image --ignore-unfixed --format table --vuln-type os,library --exit-code 1 --severity HIGH --input /tmp/${{ matrix.package }}.tar
+  
+  push:
+    name: Push to registry
+    needs: [docker-grype, docker-trivy, version-generator, node-build]
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        package: [auth, backend, frontend]
+
+    steps:
+      - name: Fetch docker image from cache
+        uses: actions/cache/restore@v4
+        with:
+          path: /tmp/${{ matrix.package }}.tar
+          key: ${{ github.sha }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Load image into Docker
+        run: |
+          docker load --input /tmp/${{ matrix.package }}.tar
+      - name: List docker images
+        run: docker images
+      - name: Push to registry
+        run: |
+          docker push ghcr.io/${{ github.repository_owner }}/rafiki-${{ matrix.package }}:${{ needs.version-generator.outputs.version }}
+  
+  generate-release:
+    runs-on: ubuntu-latest
+    needs: [push, version-generator]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Generate CHANGELOG data
+        id: changelog
+        uses: requarks/changelog-action@v1
```

Comment:
This creates the changelog based on the template, correct?

Reply:
This actually generates changelog data based on commits between current and previous same tag. It does not create changelog. If you want, I can also generate CHANGELOG.md here or append the data to it?

Reply:
Do you know how it orders the commits in the generated changelog?
Right now, we have the following defined: https://github.com/interledger/rafiki/blob/main/.github/release.yaml

Reply:
Yes, if commit has note `BREAKING CHANGE` then this breaking changes will be always on top. After that, there will be bug fixes, features, chores, etc. I just need to check for dependencies.

Reply:
@sabineschaller I'm not sure if this action takes into account the label of the PRs and the `breaking > general changes > dependency updates` order, since it checks commits and not PRs.

That being said, it does look like though we should get into a habit of doing `feat!: ...` (or adding `BREAKING CHANGE` to the to-be squashed commit), so we _could_ see & auto generate breaking changes just from the commits